Google.Api.Generator/Generation/MetadataGenerator.cs

```diff
@@ -0,0 +1,91 @@
+﻿// Copyright 2021 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using Google.Gapic.Metadata;
+using static Google.Gapic.Metadata.GapicMetadata.Types;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Google.Api.Generator.Generation
+{
+    internal static class MetadataGenerator
+    {
+        private const string GapicMetadataSchemaVersion = "1.0";
+        private const string GapicMetadataFileDescription = "This file maps proto services/RPCs to the corresponding library clients/methods";
+        private const string GapicMetadataLanguage = "csharp";
+
+        private const string TransportKeyGrpc = "grpc";
+
+        internal static string GenerateGapicMetadataJson(List<ServiceDetails> allServiceDetails)
+        {
+            var gapicMetadata = new GapicMetadata
+            {
+                Schema = GapicMetadataSchemaVersion,
+                Comment = GapicMetadataFileDescription,
+                Language = GapicMetadataLanguage,
+                ProtoPackage = allServiceDetails.First().ProtoPackage,
+                LibraryPackage = allServiceDetails.First().Namespace,
+                Services = {

```

Comment:
I'd personally suggest formatting this with the brace on the next line, indent the method call, and put the closing brace on a line on its own. (I'd probably put the method call all on one line instead of over four lines, too. But this isn't a blocker, and it's pretty subjective.

Reply:
+1

Reply:
👍